crates/compiler/parse/src/ast.rs

```diff
@@ -1183,51 +1190,39 @@ impl<'a> Defs<'a> {
         })
     }
 
-    // We could have a type annotation as the last tag,
-    // this helper ensures we refer to the last value_def
-    // and that we remove the correct tag
-    pub fn last_value_suffixed(&self) -> Option<(Self, &'a Loc<Expr<'a>>)> {
-        let value_indexes =
-            self.tags
-                .clone()
-                .into_iter()
-                .enumerate()
-                .filter_map(|(tag_index, tag)| match tag.split() {
-                    Ok(_) => None,
-                    Err(value_index) => Some((tag_index, value_index.index())),
-                });
-
-        if let Some((tag_index, value_index)) = value_indexes.last() {
-            match self.value_defs[value_index] {
-                ValueDef::Body(
-                    Loc {
-                        value: Pattern::RecordDestructure(collection),
-                        ..
-                    },
-                    loc_expr,
-                ) if collection.is_empty() && is_expr_suffixed(&loc_expr.value) => {
-                    let mut new_defs = self.clone();
-                    new_defs.remove_value_def(tag_index);
-
-                    return Some((new_defs, loc_expr));
-                }
-                ValueDef::Stmt(loc_expr) if is_expr_suffixed(&loc_expr.value) => {
-                    let mut new_defs = self.clone();
-                    new_defs.remove_value_def(tag_index);
+    pub fn pop_last_value(&mut self) -> Option<&'a Loc<Expr<'a>>> {
```

Comment:
We can return the last expression as is, so we don't need to check if it's suffixed